File: app/src/test/java/com/mgaetan89/showsrage/adapter/SearchResultsAdapter_GetItemCountTest.java
package com.mgaetan89.showsrage.adapter;

import com.mgaetan89.showsrage.model.SearchResultItem;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class SearchResultsAdapter_GetItemCountTest {
	@Parameterized.Parameter(1)
	public int itemCount;

	@Parameterized.Parameter(0)
	public List<SearchResultItem> searchResultItems;

	private SearchResultsAdapter adapter;

	@Before
	public void before() {
		this.adapter = new SearchResultsAdapter(this.searchResultItems);
	}

	@Test
	public void getItemCount() {
		assertThat(this.adapter.getItemCount()).isEqualTo(this.itemCount);
	}

	@After
	public void after() {
		this.adapter = null;
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{null, 0},
				{Collections.emptyList(), 0},
				{Collections.singletonList(new SearchResultItem()), 1},
				{Arrays.asList(new SearchResultItem(), new SearchResultItem(), new SearchResultItem()), 3},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/adapter/SearchResultsAdapter_GetItemCountTest.kt
package com.mgaetan89.showsrage.adapter

import com.mgaetan89.showsrage.model.SearchResultItem
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class SearchResultsAdapter_GetItemCountTest(val searchResultItems: List<SearchResultItem>, val itemCount: Int) {
	private lateinit var adapter: SearchResultsAdapter

	@Before
	fun before() {
		this.adapter = SearchResultsAdapter(this.searchResultItems)
	}

	@Test
	fun getItemCount() {
		assertThat(this.adapter.itemCount).isEqualTo(this.itemCount)
	}

	companion object {
		@JvmStatic
		@Parameterized.Parameters
		fun data(): Collection<Array<Any>> {
			return listOf(
					arrayOf(emptyList<Any>(), 0),
					arrayOf(listOf(SearchResultItem()), 1),
					arrayOf(listOf(SearchResultItem(), SearchResultItem(), SearchResultItem()), 3)
			)
		}
	}
}
